controllers/build.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"encoding/json"
+	"strings"
+
+	routev1 "github.com/openshift/api/route/v1"
+	appstudiov1alpha1 "github.com/redhat-appstudio/application-service/api/v1alpha1"
+	tektonapi "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	triggersapi "github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func determineBuildDefinition(component appstudiov1alpha1.Component) string {
+	return "devfile-build"
+}
+
+func determineBuildCatalog(namespace string) string {
+	// TODO: If there's a namespace/workspace specific catalog, we got
+	// to respect that.
+	return "quay.io/redhat-appstudio/build-templates-bundle:v0.1.2"
+}
+
+// determineBuildExecution returns the pipelineRun spec
+func determineBuildExecution(component appstudiov1alpha1.Component, params []tektonapi.Param, workspaceSubPath string) tektonapi.PipelineRunSpec {
+
+	pipelineRunSpec := tektonapi.PipelineRunSpec{
+		Params: params, //paramsForWebhookBasedBuilds(component),
+		PipelineRef: &tektonapi.PipelineRef{
+
+			// This can't be hardcoded to devfile-build.
+			// The logic should determine if it is a
+			// nodejs build, java build, dockerfile build or a devfile build.
```

Comment:
wondering how would this happen? Is it something that HAC would know and the component.Spec should have a field to indicate the type of build?

Reply:
@maysunfaisal Im wondering if ComponentDetectionQuery would help us with those?